### ISSUE TITLE
Discrepancy between results in v3.0.0.beta1 and v3.0.0.beta2

### ISSUE THREAD

**Issue by remusao** [imgproxy/imgproxy#805]
Hi,

First of all, thank you very much for working on imgproxy, it is an amazing project. While migrating to v3 recently, I observed a discrepancy on some images (potentially the ones involving transparency). After checking intermediary versions between the latest v2 and first v3 version, it appeared that the change was introduced in v3.0.0.beta2. I have created some repro steps to ease the investigation: https://github.com/remusao/imgproxy-issue-repro

It is unclear to me if this change is something that was expected while migrating from v2 to v3 and might require a different configuration or if this is a regression. Here is one image where the issue can be observed:

[Original image](https://cdn.sstatic.net/Sites/stackoverflow/Img/favicon.ico?v=ec617d715196): ![image](https://user-images.githubusercontent.com/1299873/155754526-4710bbfd-4814-47af-af59-c6a6cc06a075.png)

Resized with v3.0.0.beta1: ![image](https://user-images.githubusercontent.com/1299873/155754470-11f5b0d8-f3f4-4928-9355-60624e50758f.png)

Resized with v3.0.0.beta2: ![image](https://user-images.githubusercontent.com/1299873/155754606-5bf3bed3-57cd-41b2-ad8a-f892b869b9a8.png)

Minimum docker-compose.yml to start locally:
```yaml
version: "3"

services:
  imgproxy:
    image: "darthsim/imgproxy:v3.0.0.${VERSION}"
    environment:
      - IMGPROXY_BIND=0.0.0.0:8000
      - IMGPROXY_KEY=943b421c9eb07c830af81030552c86009268de4e532ba2ee2eab8247c6da0881
      - IMGPROXY_SALT=520f986b998545b4785e0defbc4f3c1203f22de2374a3d53cb7a7fe9fea309c5
    ports:
      - 127.0.0.1:8000:8000
```

Starting with v3.0.0.beta1: `VERSION=beta1 docker-compose up --build`
Starting with v3.0.0.beta2: `VERSION=beta2 docker-compose up --build`

Visit the following URL with both version to see the difference: `http://localhost:8000/fSeKiHqtklTyA0CDo_BkvOLPq2-r08PzzpfBjWO3lfs/rs:fill:32:32:1/g:no/aHR0cHM6Ly9jZG4u/c3N0YXRpYy5uZXQv/U2l0ZXMvc3RhY2tv/dmVyZmxvdy9JbWcv/ZmF2aWNvbi5pY28_/dj1lYzYxN2Q3MTUx/OTY.png`

The diff between both versions: https://github.com/imgproxy/imgproxy/compare/v3.0.0.beta1...v3.0.0.beta2

Thanks in advance for the help and insights,

**Comment by DarthSim** [comment]
Hey @remusao!
Thanks for catching this! Fixed in `v3.3.1`.

**Comment by remusao** [comment]
Thanks @DarthSim, I can confirm that the issue is fixed in `v3.3.1`.
